compiler/qsc_codegen/src/qir_base.rs

```diff
@@ -169,21 +179,28 @@ impl BaseProfSim {
         id
     }
 
-    fn write_output_recording(&mut self, val: &Value) -> std::fmt::Result {
+    fn write_output_recording(
+        &mut self,
+        val: &Value,
+        entry_span: PackageSpan,
+    ) -> std::result::Result<(), (Error, Vec<Frame>)> {
         match val {
             Value::Array(arr) => {
-                self.write_array_recording(arr.len())?;
+                if arr.is_empty() {
+                    return Err((qsc_eval::Error::EmptyArrayOutput(entry_span), Vec::new()));
```

Comment:
It would be nice to return a real call stack here, no? Is there any way you can use `get_stack_frames` from here?

Reply:
That's the tricky thing... we aren't inside of execution anymore, we are in the post-execution output processing. This code doesn't run "inside" any callable or even in the entry expression, it's something we do with the result after execution is done. So empty call stack is technically correct.